client/extensions/woocommerce/app/order/order-payment/dialog.js

```diff
@@ -0,0 +1,226 @@
+	};
+
+	recalculateRefund = total => {
+		this.setState( { refundTotal: total } );
+	};
+
+	updateNote = event => {
+		this.setState( {
+			refundNote: event.target.value,
+		} );
+	};
+
+	sendRefund = () => {
+		const { order, paymentMethod, siteId, translate } = this.props;
+		const maxRefund = parseFloat( order.total ) + getOrderRefundTotal( order );
+		if ( this.state.refundTotal > maxRefund ) {
+			this.setState( {
+				errorMessage: translate( 'Refund must be less than or equal to the order total.' ),
+			} );
+			return;
+		} else if ( this.state.refundTotal <= 0 ) {
+			this.setState( { errorMessage: translate( 'Refund must be greater than zero.' ) } );
+			return;
+		}
+		this.toggleDialog();
+		const refundObj = {
+			amount: this.state.refundTotal + '', // API expects a string
+			reason: this.state.refundNote,
+			api_refund: paymentMethod && -1 !== paymentMethod.method_supports.indexOf( 'refunds' ),
+		};
+		this.props.sendRefund( siteId, order.id, refundObj );
+	};
+
+	renderCreditCard = () => {
+		const { isPaymentLoading, paymentMethod, translate } = this.props;
+		if ( isPaymentLoading ) {
+			return null;
+		}
+
+		if ( paymentMethod && -1 === paymentMethod.method_supports.indexOf( 'refunds' ) ) {
+			return (
+				<div className="order-payment__method">
+					<h3>{ translate( 'Manual Refund' ) }</h3>
+					<p>
+						{ translate(
+							"This payment method doesn't support automated refunds and must be submitted manually."
+						) }
+					</p>
+				</div>
+			);
+		}
+
+		return (
+			<div className="order-payment__method">
+				<h3>
+					{ translate( 'Refunding payment via %(method)s', {
+						args: {
+							method: paymentMethod.title,
+						},
+					} ) }
+				</h3>
+			</div>
+		);
+	};
+
+	render() {
+		const { isPaymentLoading, order, isVisible, translate } = this.props;
+		const { errorMessage, refundNote } = this.state;
+		const dialogClass = 'woocommerce'; // eslint/css specificity hack
+
+		if ( 'cancelled' === order.status || 'failed' === order.status ) {
+			return null;
+		}
+
+		let refundTotal = formatCurrency( 0, order.currency );
+		if ( this.state.refundTotal ) {
+			refundTotal = formatCurrency( this.state.refundTotal, order.currency );
+		}
+		refundTotal = refundTotal.replace( /[^0-9.,]/g, '' );
+
+		const dialogButtons = [
+			<Button onClick={ this.toggleDialog }>{ translate( 'Cancel' ) }</Button>,
+			<Button primary onClick={ this.sendRefund } disabled={ isPaymentLoading }>
+				{ translate( 'Refund' ) }
+			</Button>,
+		];
+
+		return (
+			<Dialog
+				isVisible={ isVisible }
+				onClose={ this.toggleDialog }
+				className={ dialogClass }
+				buttons={ dialogButtons }
+				additionalClassNames="order-payment__dialog woocommerce"
+			>
+				<h1>{ translate( 'Refund order' ) }</h1>
+				<OrderRefundTable order={ order } onChange={ this.recalculateRefund } />
+				<form className="order-payment__container">
+					<FormLabel className="order-payment__note">
+						{ translate( 'Refund note' ) }
```

Comment:
Please add a little context for the translators here, since 'note' could possibly be mistaken for currency when it appears with 'Refund.'